webhook/resources/objectstore/mutator.go

```diff
@@ -0,0 +1,68 @@
+package objectstore
+
+import (
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	"github.com/longhorn/longhorn-manager/webhook/common"
+	"github.com/pkg/errors"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type objectStoreMutator struct {
+	admission.DefaultMutator
+	ds *datastore.DataStore
+}
+
+func NewMutator(ds *datastore.DataStore) admission.Mutator {
+	return &objectStoreMutator{ds: ds}
+}
+
+func (osm *objectStoreMutator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "objectstores",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.ObjectStore{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (osm *objectStoreMutator) Create(req *admission.Request, obj runtime.Object) (ops admission.PatchOps, err error) {
+	store, ops, err := getDefaultMutations(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if store.Spec.TargetState == "" {
+		op := "{\"op\": \"add\", \"path\": \"/spec/targetState\", \"value\": \"running\"}"
+		ops = append(ops, op)
+	}
```

Comment:
Suggesting to set the default s3gwImage and uiImage here instead of setting it inside the ObjectStore controller